server/internal/core/application/covenantless.go

```diff
@@ -1464,8 +1419,102 @@ func (s *covenantlessService) saveEvents(
 	return s.repoManager.Rounds().AddOrUpdateRound(ctx, *round)
 }
 
+func (s *covenantlessService) markAsRedeemed(ctx context.Context, vtxo domain.Vtxo) {
```

Comment:
for sake of consistency, this should be moved after `reactToFraud`. Or either in covenant move it before `reactToFraud`, as you prefer